bindings/rust/wasmedge-sys/src/vm.rs

```diff
@@ -74,13 +75,69 @@ impl Vm {
         Ok(self)
     }
 
+    /// Instantiate the validated WASM module in the VM context.
+    ///
+    /// This is the third step to invoke a WASM function step by step.
+    /// After validating a WASM module in the VM context, You can call this function
+    /// to instantiate it. And you can then call `execute` for invoking
+    /// the exported function in this WASM module.
+    ///
     pub fn instantiate(self) -> WasmEdgeResult<Self> {
         unsafe {
             check(wasmedge::WasmEdge_VMInstantiate(self.ctx))?;
         }
         Ok(self)
     }
 
+    /// Get the function type by function name.
+    pub fn func_type(&self, func_name: impl AsRef<str>) -> Option<FuncType> {
```

Comment:
According to the hierarchy of the WebAssebly concepts, the "Contains" relationship exists between VM and Store, and between Store and Function. However, the form of `vm.functions` implies such a direct "Contains" relationship between VM and Function. That could lead users to misunderstand the hierarchy model of the concepts. In addition, the current design is concise and easier to maintain for the future. Therefore, I suggest keeping the current design.

Reply:
Okay, it may be good to use `vm.get_func_type(name)` here, because the concept that function is not contained by VM, and we also avoid using  `vm.func_type(name)` with a misleading concept.

BTW, the registered one should be `vm.get_registered_func_type`.

Reply:
Accepted. I'll refactor these APIs to follow your comment. Thanks a lot!